src/stream.rs

```diff
@@ -0,0 +1,58 @@
+use std::io;
+use std::marker::PhantomData;
+use Event;
+
+pub struct HtmlStreamer<'a, I> {
+    ev_iter: I,
+    ev_life: PhantomData<Event<'a>>,
+}
+
+impl<'a, I> HtmlStreamer<'a, I>
+    where I: Iterator<Item = Event<'a>>
+{
+    pub fn new<T>(events: T) -> HtmlStreamer<'a, I>
+        where T: IntoIterator<IntoIter = I, Item = Event<'a>> + 'a
+    {
+        HtmlStreamer {
+            ev_iter: events.into_iter(),
+            ev_life: PhantomData,
+        }
+    }
+
+    pub fn stream(self, w: &mut io::Write) -> io::Result<usize> {
+        for ev in self.ev_iter {
+            try!(write!(w, "{}", ev));
+        }
+        return Ok(0);
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use HtmlStreamer;
+    use Event;
+
+    #[test]
+    fn test() {
+        let events = vec![
+            Event::start_tag("h1", attr_set!(id="hello", class="fun"), false),
+            Event::text("Hello, "),
+            Event::raw_html(""), // empty event
+            Event::start_tag("small", attr_set!(), false),
+            Event::text("world"),
+            Event::end_tag("small"),
+            Event::start_tag("img", attr_set!(src="foo-link"), true),
+            Event::start_tag("br", attr_set!(dataAttr="'1'"), true),
+            Event::end_tag("h1"),
```

Comment:
:+1: I like the methods instead of macros.

Could consider not having the `self_closing` flag in the `start_tag` method, it's definitely the minority case and I think having a method to set it is more self-documenting (at least until rust gets keyword arguments somehow).